server/src/controllers/grades.ts

```diff
@@ -758,8 +758,21 @@ export function parseGradesFromCsv(
  * the CSV file contains attainments which don't belong to the specified course or course instance.
  */
 export async function addGrades(req: Request, res: Response, next: NextFunction): Promise<void> {
-  /** TODO: Check grading points are not higher than max points of the attainment. */
+  const requestSchema: yup.AnyObjectSchema = yup.object().shape({
+    completionDate: yup
+      .date()
+      .notRequired(),
```

Comment:
This should not be optional, as it is essential information to know when the grades are no longer valid.

Reply:
marked as required, both back and frontend